airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -226,33 +226,29 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs: Any) -
             ...]
         """
 
+        today = pendulum.now().date()
         start_date = pendulum.parse(self.start_date).date()
-        end_date = pendulum.now().date()
-
-        # Determine stream_state, if no stream_state we use start_date
         if stream_state:
-            start_date = pendulum.parse(stream_state.get(self.cursor_field)).date()
+            prev_end_date = pendulum.parse(stream_state.get(self.cursor_field)).date()
+            start_date = prev_end_date.add(days=1)
+        end_date = today
+        if start_date > end_date:
+            return [None]
 
-        # use the lowest date between start_date and self.end_date, otherwise API fails if start_date is in future
-        start_date = min(start_date, end_date)
         date_slices = []
-        while start_date <= end_date:
-            end_date_slice = start_date.add(days=self.window_in_days)
+        slice_start_date, slice_end_date = start_date, None
```

Comment:
Do we need to define the `slice_end_date` here?
Because we setting it later inside the loop:
`slice_end_date = slice_start_date.add(days=self.window_in_days)`

Reply:
yep, no need. removed it